### PR TITLE
[pcre2] Fix CMake integration.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
This PR applies the patch in PCRE2Project/pcre2#260 to fix the previously broken CMake integration.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Please mark this as draft.

**Comment by dg0yt** [comment]
I opened #31929 now to ensure that the pristine update doesn't cause any regressions.

I realize now that the CMake config may be underused (or even unused) in vcpkg (or in general) because we didn't install it due to its poor state. So this PR will probably not give the insight I was hoping for.

**Comment by teo-tsirpanis** [comment]
Some ports break because they were searching for the PCRE2 library in `lib`. 😭 https://github.com/pocoproject/poco/blob/a00bfbe89fccabbe3606b31c0a1c7edd3da14e2e/cmake/FindPCRE2.cmake#L69-L77

**Comment by dg0yt** [comment]
> Some ports break because they were searching for the PCRE2 library in `lib`. sob https://github.com/pocoproject/poco/blob/a00bfbe89fccabbe3606b31c0a1c7edd3da14e2e/cmake/FindPCRE2.cmake#L69-L77

`find_library` breaking in *vendored* Find modules is unrelated to adding *official* exported CMake config to pcre2.
Port poco doesn't even use this Find module, see patch `ports/poco/0007-find-pcre2.patch`.

File: ports/pcre2/portfile.cmake [hunk 1]
```diff
@@ -31,35 +32,24 @@ vcpkg_cmake_configure(
...
 vcpkg_copy_pdbs()
-
-file(READ "${CURRENT_PACKAGES_DIR}/include/pcre2.h" PCRE2_H)
-if(BUILD_STATIC)
-    string(REPLACE "defined(PCRE2_STATIC)" "1" PCRE2_H "${PCRE2_H}")
-else()
-    string(REPLACE "defined(PCRE2_STATIC)" "0" PCRE2_H "${PCRE2_H}")
-endif()
-file(WRITE "${CURRENT_PACKAGES_DIR}/include/pcre2.h" "${PCRE2_H}")
-
```
Comment (dg0yt) [inline ports/pcre2/portfile.cmake:51]: Don't remove this. It causes the build errors. Nobody is using the new CMake config, and some will never do.

Comment (teo-tsirpanis) [reply ports/pcre2/portfile.cmake]: Fixed. `qt5-websockets` fails, will investigate tomorrow. And I will also rebase the PR.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Thanks for your PR. PR https://github.com/microsoft/vcpkg/pull/31929 did update `pcre2` to 10.42, it has been merged, please resolve the conflicts.

**Comment by SamuelMarks** [comment]
Any progress here? - Really need PCRE2 for my project (porting libbsd to Windows)

**Comment by teo-tsirpanis** [comment]
I am waiting for it to get merged upstream in  PCREProject/pcre2#260.

File: ports/pcre2/portfile.cmake (hunk 1 above)
Comment (BurningEnlightenment) [reply ports/pcre2/portfile.cmake:51]: You can refactor this to use [`vcpkg_replace_string`](https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_replace_string)

File: ports/pcre2/portfile.cmake [hunk 2]
```diff
@@ -31,11 +32,11 @@ vcpkg_cmake_configure(
         -DPCRE2_SUPPORT_UNICODE=ON
         -DPCRE2_BUILD_TESTS=OFF
         -DPCRE2_BUILD_PCRE2GREP=OFF
+        -DCMAKE_DISABLE_FIND_PACKAGE_BZip2=ON
```
Comment (BillyONeal) [inline ports/pcre2/portfile.cmake:35]: Can you explain what this part of the change is doing?

**Review by BillyONeal — COMMENTED** [review]
I fixed the problem that caused every line to be edited; there was an accidental LF -> CRLF in there.

File: ports/pcre2/usage [hunk 3]
```diff
@@ -0,0 +1,6 @@
+The package pcre2 is compatible with built-in CMake targets:
+
+    # Each component imports a target:
+    # TARGETS: pcre2::8BIT pcre2::16BIT pcre2::32BIT pcre2::POSIX
+    find_package(pcre2 CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE pcre2::8BIT pcre2::POSIX)
```
Comment (LilyWangLL) [inline ports/pcre2/usage:6]: Usage test failed with the following error:
```
1> [CMake] CMake Error at PRreview/CMakeLists.txt:14 (target_link_libraries):
1> [CMake]   Target "PRreview" links to:
1> [CMake]     pcre2::8BIT
1> [CMake]     pcre2::POSIX
1> [CMake] 
1> [CMake]   but the target was not found.
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/pcre2/portfile.cmake (hunk 2 above)
Comment (teo-tsirpanis) [reply ports/pcre2/portfile.cmake:35]: BZip2 and Zlib are used by the `pcre2grep` program which was disabled by the line just above.

File: ports/pcre2/usage (hunk 3 above)
Comment (teo-tsirpanis) [reply ports/pcre2/usage]: `pcre2` was in all caps, fixed, let's see…

**Review by LilyWangLL — APPROVED** [review]

**Comment by teo-tsirpanis** [comment]
Can we merge it?

**Pull request merged**